src/main/java/com/cloudbees/jenkins/plugins/bitbucket/client/Cache.java

```diff
@@ -0,0 +1,62 @@
+package com.cloudbees.jenkins.plugins.bitbucket.client;
+
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+public class Cache<K, V> extends LinkedHashMap<K, V> {
+
+    private static final int MAX_ENTRIES = 100;
+
+    private final Map<K, Long> entryDates = new HashMap<>();
+
+    private final long expireAfterNanos;
+
+    public Cache(final int duration, final TimeUnit unit) {
+        super(16, 0.75F, true);
+        this.expireAfterNanos = unit.toNanos(duration);
+    }
+
+    @Override
+    protected boolean removeEldestEntry(final Map.Entry<K, V> eldest) {
+        return size() > MAX_ENTRIES;
+    }
+
+    public synchronized V get(final K key, final Callable<V> callable) throws ExecutionException {
+        if (isExpired(key)) {
+            doRemove(key);
+        }
+
+        if (containsKey(key)) {
+            return get(key);
+        }
+
+        V result;
+        try {
+            result = callable.call();
+        } catch (final Exception e) {
+            throw new ExecutionException("Cannot load value for key: " + key, e);
+        }
+
+        return doPut(key, result);
+    }
+
+    private boolean isExpired(final K key) {
+        return entryDates.get(key) != null && System.nanoTime() - entryDates.get(key) > expireAfterNanos;
```

Comment:
extract the `get` call into a local variable store.

Also if this is the logic you are using then you probably do not want the cache using last access... which might point to a tighter cache implementation (namely wrapping V in a container that holds the entry date and changing from extending LinkedHashMap to using the map internally and only exposing the get method via delegation with unwrap)